docs/madr/decisions/073-spiffe-compliance.md

```diff
@@ -0,0 +1,136 @@
+# SPIFFE Compliance
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/13659
+
+## Context and Problem Statement
+
+In today’s service mesh systems, secure communication between services is essential. To make that possible, each service needs a clear and trusted identity: so we know exactly who is talking to whom.
+
+One of the most trusted standards for defining these service identities is SPIFFE (Secure Production Identity Framework for Everyone). It gives us:
+
+* A standard way to name services: `spiffe://<trust-domain>/<path>`
+* A strict format for certificates, especially the Subject Alternative Name (SAN) - which must contain exactly one URI (the SPIFFE ID)
+* Tools and rules for issuing, rotating, and checking these identities
+* SPIFFE certificates identify workload, not an endpoint or API
+
+Right now, Kuma does something similar by generating SPIFFE-like IDs like `spiffe://<mesh-name>/<service-name>`. But it also adds extra SANs (like `kuma://kuma.io/version/v1`) based on tags. These extra SANs make Kuma's certificates non-compliant with SPIFFE, meaning they can’t be used directly with SPIFFE-based systems like SPIRE, or other tools that expect full compliance. Also, Kuma uses certificates to identify endpoints, whereas SPIFFE-based certificates are used to establish identity via SPIFFE IDs of the workload.
+
+In SPIFFE, a Trust Domain is a logical boundary for identity and trust, similar to a DNS domain. Services within the same trust domain can securely identify and trust each other. Communication across different trust domains (such as between organizations or clusters) requires federation, which establishes trust relationships so that identities from one trust domain are recognized and trusted by another.
+
+Kuma currently uses the mesh name as the trust domain. While this works inside Kuma, it doesn’t clearly separate trust boundaries, and it limits how easily Kuma can connect with external services or other applications in the same trust domain.
+
+This setup causes problems when:
+
+* You want to connect Kuma with other SPIFFE-compliant systems outside the mesh
+* You run multiple clusters that need to trust each other
+* You work in environments that require strict identity standards for security or compliance reasons
+
+To fully support SPIFFE and modern security practices, Kuma should:
+
+* Issue certificates that include only one SPIFFE-compliant URI SAN (currently we generate `spiffe://<mesh>/<service>` for each inbound and `kuma://kuma.io/<tag_name>/<tag_value>` for each tag)
+* Let users set the trust domain explicitly, not tie it to the mesh name
+* Allow using other systems (like SPIRE) to manage identities
```

Comment:
I think one very important thing:

- Kuma should separate issuing identity from validating trust.